GitUI/CommandsDialogs/RepoHosting/CreatePullRequestForm.cs

```diff
@@ -112,6 +115,26 @@ private void _pullReqTargetsCB_SelectedIndexChanged(object sender, EventArgs e)
         [CanBeNull]
         private IHostedRemote MyRemote => _hostedRemotes.FirstOrDefault(r => r.IsOwnedByMe);
 
+        private void LoadPRTemplate()
+        {
```

Comment:
There is a room for improvement - the operation must not be blocking, all IO must happen on a background thread. 
Please have a look how we download avatars:
https://github.com/gitextensions/gitextensions/blob/a177c46a2f0e7a42481d21e2f5638d9d610e1b95/GitUI/Avatars/AvatarDownloader.cs#L153-L187
I think for the most part you can reuse the code (you won't need semaphores).

Have you thought of failure scenarios? E.g. 
* network is unavailable
* timeouts
* HTTP 404
* HTTP 500, etc

What is a proposal for UX?